### PR TITLE
Check and cache nonces when verifying signatures

### PR DESCRIPTION
**Pull request by shreyav**

### CODE REVIEW

File: uma/nonce_cache.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Copyright ©, 2022-present, Lightspark Group, Inc. - All Rights Reserved
+
+from abc import ABC, abstractmethod
+from typing import Dict
+from uma.exceptions import InvalidNonceException
+
+
+class INonceCache(ABC):
+    @abstractmethod
+    def check_and_save_nonce(self, nonce: str, timestamp: int) -> None:
```
Comment (jklein24) [inline uma/nonce_cache.py:10]: Would it make sense to take a datetime here instead of an int?

**Review by jklein24 — APPROVED** [review]

File: uma/nonce_cache.py (hunk 1 above)
Comment (shreyav) [reply uma/nonce_cache.py:10]: yeah, that sounds good -- the request/response timestamps are inconsistent (some use datetime, some use int), maybe we should also unify them to all use datetime?

Comment (jklein24) [reply uma/nonce_cache.py:10]: Hmm yeah might be worthwhile. My fault!

Comment (shreyav) [reply uma/nonce_cache.py]: sg! will do that in v1.0

**Review by jklein24 — APPROVED** [review]

**Pull request merged**